benchmarks/oss_fuzz_benchmark_integration.py

```diff
@@ -0,0 +1,164 @@
+import argparse
+import datetime
+from distutils import dir_util
+import os
+import sys
+
+from common import utils
+
+# pytype: disable=import-error
+# pylint: disable=import-error,wrong-import-position,ungrouped-imports,too-many-arguments
+
+# TODO(metzman): Don't rely on OSS-Fuzz code. We don't want to depend on it
+# because it can easily break use. Especially becuase:
+# 1. We use private methods
+# 2. The OSS-Fuzz code depends on absolute imports.
+# 3. The OSS-Fuzz code assumes it is run from the OSS-Fuzz directory and can
+# accdidentaly break our repo.
+OSS_FUZZ_REPO_PATH = os.path.join(utils.ROOT_DIR, 'third_party', 'oss-fuzz',
+                                  'infra')
+sys.path.append(OSS_FUZZ_REPO_PATH)
+
+import bisector
+import build_specified_commit
+import helper
+import repo_manager
+
+from common import logs
+from common import yaml_utils
+
+
+def copy_dir_contents(src_dir, dst_dir):
+    """Copy the contents of |src_dir| into |dst_dir|."""
+    return dir_util.copy_tree(src_dir, dst_dir)
+
+
+def copy_oss_fuzz_files(project, commit_date, benchmark_dir):
+    """Checkout the right files from OSS-Fuzz to build the benchmark based on
+    |project| and |commit_date|. Then copy them to |benchmark_dir|."""
+    cwd = os.getcwd()
+    oss_fuzz_repo_manager = repo_manager.BaseRepoManager(helper.OSS_FUZZ_DIR)
+    try:
+        projects_dir = os.path.join(helper.OSS_FUZZ_DIR, 'projects', project)
+        os.chdir(helper.OSS_FUZZ_DIR)
+        oss_fuzz_commit, _, _ = oss_fuzz_repo_manager.git([
+            'log', '--before=' + commit_date.isoformat(), '-n1', '--format=%H',
+            projects_dir
+        ],
+                                                          check_result=True)
+        oss_fuzz_commit = oss_fuzz_commit.strip()
+        if not oss_fuzz_commit:
+            logs.warning('No suitable earlier OSS-Fuzz commit found.')
+            return False
+        oss_fuzz_repo_manager.git(['checkout', oss_fuzz_commit, projects_dir],
+                                  check_result=True)
+        copy_dir_contents(projects_dir, benchmark_dir)
+        return True
+    finally:
+        oss_fuzz_repo_manager.git(['reset', '--hard'])
+        # This must be done in this order or else we reset our fuzzbench repo.
+        os.chdir(cwd)
+
+
+def get_benchmark_name(project, fuzz_target, benchmark_name=None):
+    """Returns the name of the benchmark. Returns |benchmark_name| if is set.
+    Otherwise returns a name based on |project| and |fuzz_target|."""
+    if benchmark_name is not None:
+        return benchmark_name
+    return project + '_' + fuzz_target
+
+
+def replace_base_builder(benchmark_dir, commit_date):
+    """Replace the parent image of the Dockerfile in |benchmark_dir|,
+    base-builder (latest), with a version of base-builder that is likely to
+    build the project as it was on |commit_date| without issue."""
+    base_builder_repo = bisector._load_base_builder_repo()  # pylint: disable=protected-access
```

Comment:
Just making sure that the image which this is being run on has gcloud installed?

Reply:
Sorry if I misunderstand the question, but if you mean local test for running the integration, yes it was tested with gcloud installed.

Reply:
>Just making sure that the image which this is being run on has gcloud installed?

This is run on developers hosts so actually we don't know that gcloud is installed. We could add it as a project wide dependency or check if it is installed in this script and if it isn't point to the installation instructions.
I favor this over adding it as a project-wide dependency since I'm not 100% sure we want to keep this dependency because I'm not sure we want to use the "right" OSS-Fuzz builder image instead of "latest", since at this point it's one of the only things seperating OSS-Fuzz benchmarks from standard ones. If we get rid of this it may cause some benchmarks not to build but will simplify things. I will weight these tradeoffs at a later time.